dll/win32/shimgvw/shimgvw.c

```diff
@@ -435,8 +435,9 @@ static void pSaveImageAs(HWND hwnd)
     sfn.hInstance   = hInstance;
     sfn.lpstrFile   = szSaveFileName;
     sfn.lpstrFilter = szFilterMask;
-    sfn.nMaxFile    = MAX_PATH;
+    sfn.nMaxFile    = _countof(szSaveFileName);
     sfn.Flags       = OFN_EXPLORER | OFN_OVERWRITEPROMPT | OFN_HIDEREADONLY;
+    sfn.lpstrDefExt = L"png";
```

Comment:
I am not sure whether that is the correct solution.
If we open a bmp, it should stick with bmp.
If we open a jpg, it should stick with jpg.

It should not change the extension, but "try to preserve the format the was opened". I tested behavior of MS previewer on XPSP3 to come to that conclusion.

Reply:
This member is used just to decide whether the app should append a filename extension or not. The type of the extension is not important.

It works also for `.jpg` and `.bmp`.

Reply:
Thank you for the explanation. Resolved then.